regression/cbmc-cover/block-coverage-report2/test.desc

```diff
@@ -0,0 +1,12 @@
+CORE
+main.c
+--cover location
+block 1 \(lines main.c:main:13,14\): SATISFIED
+block 2 \(lines main.c:main:15\): SATISFIED
+block 3 \(lines main.c:main:17,18\): SATISFIED
+block 4 \(lines main.c:main:18,19\): SATISFIED
+block 5 \(lines main.c:main:20\): SATISFIED
+block 6 \(lines main.c:main:15,21,22\): SATISFIED
+block 7 \(lines main.c:main:24,25\): SATISFIED
+block 1 \(lines main.c:foo:6,7,8,9\): SATISFIED
```

Comment:
Yes, we should do that. As Mark explained in #1439 we have a tool (to be open sourced) that uses this information, so I'll first make sure it can cope with both the explicit enumeration as well as the ranges.

Reply:
Done.

Reply:
In practice, blocks are so small it makes almost no difference.  

But the essential thing is to continue listing filename-numberset pairs, because the lines contributing to a block can come from more than one file.

Reply:
That should be preserved, yes. The only change (and really the only change that was necessary to the regression tests) is that instead of `6,7,8,9` we now print `6-9`.